packages/e2e-tests/specs/editor/various/rich-text.test.js

```diff
@@ -451,4 +451,19 @@ describe( 'RichText', () => {
 		await page.keyboard.press( 'ArrowLeft' );
 		expect( await page.$( blockToolbarSelector ) ).toBe( null );
 	} );
+
+	it( 'should run input rules after composition end', async () => {
```

Comment:
I recommend adding a comment here about Puppeteer's lack of support for composition events and a link to https://github.com/puppeteer/puppeteer/issues/4981, so that we can track it and as justification for the fact that this isn't a user-centred test.